features/lorawan/system/lorawan_data_structures.h

```diff
@@ -1216,14 +1092,13 @@ typedef struct {
  * LoRaMac internal states
  */
 typedef enum {
-    LORAMAC_IDLE          = 0x00000000,
-    LORAMAC_TX_RUNNING    = 0x00000001,
-    LORAMAC_RX            = 0x00000002,
-    LORAMAC_ACK_REQ       = 0x00000004,
-    LORAMAC_ACK_RETRY     = 0x00000008,
-    LORAMAC_TX_DELAYED    = 0x00000010,
-    LORAMAC_TX_CONFIG     = 0x00000020,
-    LORAMAC_RX_ABORT      = 0x00000040,
+    LORAMAC_IDLE                    = 0x00000000,
+    LORAMAC_TX_ONGOING              = 0x00000001,
+    LORAMAC_MSG_RECVD               = 0x00000002,
+    LORAMAC_BUSY                    = 0x00000004,
+    LORAMAC_CONNECTED               = 0x00000008,
+    LORAMAC_USE_OTAA                = 0x00000010,
+    LORAMAC_TX_DONE                 = 0x00000020,
```

Comment:
Mixing flags and enums is always a bit questionable, and here you don't seem to be using them as flags at all - no combinations? But I see you clearing bits and setting bits.

If this is supposed to actually be an enum of states, then probably a good idea to not make it look like a bitfield - get rid of the values, and stop doing bit manipulations on it.

Or if it is supposed to be bits, get rid of the enum, and actually use a bitfield, as you are always manipulating bits individually, and don't need any tricky masking operations.

Maybe it's a mixture of a state and some flags.

Reply:
It is a bit field, members of the bitfield are identified as an enum. Actual storage is as a 32 bit unsigned bitfield.